components/scream/src/share/io/tests/io.cpp

```diff
@@ -501,5 +661,41 @@ get_diagnostic_input(const ekat::Comm& comm, const std::shared_ptr<GridsManager>
 
   return f_diag_h;
 }
-
+/*========================================================================================================*/
+int get_current_t(const int tt, const int dt, const int freq, const std::string& frequency_units) {
+      if (frequency_units == "nsteps") {
+        // Just use the nsteps_since_last_write information
+        return tt*dt;
+      // We will need to use timestamp information
+      } else if (frequency_units == "nsecs") {
+        return tt*freq;
+      } else if (frequency_units == "nmins") {
+        return tt*freq*60;
+      } else if (frequency_units == "nhours") {
+        return tt*freq*3600;
+      } else if (frequency_units == "ndays") {
+        return tt*freq*3600*24;
+      }
+  EKAT_REQUIRE_MSG(false,"Error: unknown frequency unit passed to get_current_t");
+}
+/*========================================================================================================*/
+TEST_CASE("input_output_basic","io")
+{
+  std::vector<std::string> output_types =
+    { "instant","average", "max", "min"};
+  std::vector<std::string> output_freq =
+    { "nsteps", "nsecs", "nmins", "nhours", "ndays"};
+  for (const auto& of : output_freq) {
+    printf("Testing output freq %s\n",of.c_str());
```

Comment:
Would you mind wrapping these in some if statement, so that only one rank prints them?